doc/platform_api/CMIS_Diagnostic_Monitoring_Overview_in_SONiC.md

```diff
@@ -0,0 +1,2375 @@
+
+The `SfpStateUpdateTask` thread is responsible for updating the static diagnostic information for all the transceivers in the system. The static diagnostic information, such as threshold values for DOM, VDM and PM, are read from the transceiver and updated in the `redis-db` during `xcvrd` boot-up and during transceiver removal and insertion.
+
+The following tables are updated by the `SfpStateUpdateTask` thread:
+
+1. `TRANSCEIVER_DOM_THRESHOLD`
+2. `TRANSCEIVER_VDM_XXX_THRESHOLD` where `XXX` is the threshold type (`highalarm`, `highwarning`, `lowwarning`, `lowalarm`)
+
+### 4.2 Dynamic Diagnostic Information
+
+The `DomInfoUpdateTask` thread is responsible for updating the dynamic diagnostic information for all the transceivers in the system. The following events drive the dynamic update of the diagnostic information:
+
+1. **Periodic update of the diagnostic information:**
+    - The `DomInfoUpdateTask` thread periodically updates the diagnostic information for all the ports.
+    - The update period interval can be retrieved by reading the `dom_info_update_periodic_secs` field from the `/usr/share/sonic/device/{platform}/{hwsku}/pmon_daemon_control.json` file.
+    - If this field or the file is absent, the default timer value is 0 seconds.
+
+2. **Link change event:**
+    - Only the **flag-related diagnostic information** is updated for a port when a link change event is detected by the `DomInfoUpdateTask` thread. Further details on the tables updated during a link change event are provided in the `Diagnostic Information Update During Link Change Event` section.
+    - Updating flag information during a link change event ensures that the flag change time is captured in a timely manner. The periodic update can take more time to update the diagnostic information since it reads the diagnostic information for all the ports in a sequential manner.
+    - Since the flag registers are clear-on-read latched values, the `DomInfoUpdateTask` thread will require two reads to update the flag value, last clear time, and change count once the flagged condition is no longer present. Hence, it is expected that the flag status change in the database will be delayed by two update cycles when the flagged condition is no longer present on the module.
+
+#### 4.2.1 High-Level Steps for Updating Dynamic Diagnostic Information
+
+1. The `DomInfoUpdateTask` thread is created by the `xcvrd` process.
+2. The `dom_info_update_periodic_secs` value is retrieved from the `pmon_daemon_control.json` file to determine the interval for updating the diagnostic information for all the ports. If the `dom_info_update_periodic_secs` field is absent or set to 0, the diagnostic information will be updated continuously without any delay between updates.
+3. The `DomInfoUpdateTask` thread starts polling for the diagnostic information of the ports in a sequential manner:
+    - It first checks if the `dom_info_update_periodic_secs` value is set. If not, it defaults to 0 seconds.
```

Comment:
@mihirpat1 can you add a section of how Xcvrd detects link event happened?

Reply:
@prgeor I have added a section for this now.